### PR TITLE
chore: refactor recent and favourite dapps to use ids

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

Use the dappsList as the source of truth for recent and favourited dapps. There was already a code comment left ages ago to make this change and Jacob mentioned it again here https://github.com/valora-inc/wallet/pull/3243#pullrequestreview-1218173556, so i thought i'd just bite the bullet.

### Other changes

n/a

### Tested

n/a

### How others should test

n/a

### Related issues

- Fixes RET-522

### Backwards compatibility

Y

### CODE REVIEW

File: src/dapps/selectors.ts [hunk 1]
```diff
@@ -60,3 +60,17 @@ export const dappCategoriesByIdSelector = createSelector(
...
+  dappsListSelector,
+  recentDappsIdsSelector,
+  (dapps, recentDappIds) => {
+    return recentDappIds.reduce((acc, recentDappId) => {
+      const recentDapp = dapps.find((dapp) => dapp.id === recentDappId)
+      if (recentDapp) {
+        return [...acc, recentDapp]
+      }
+      return acc
+    }, [] as Dapp[])
```
Comment (jeanregisser) [inline src/dapps/selectors.ts:74]: nit: I find `reduce` always hard to read compared to a plain old loop :D

File: src/dapps/slice.ts [hunk 2]
```diff
@@ -8,16 +8,15 @@ export interface State {
   dappsWebViewEnabled: boolean
   activeDapp: ActiveDapp | null
   maxNumRecentDapps: number
-  recentDapps: Dapp[]
+  recentDapps: string[]
```
Comment (jeanregisser) [inline src/dapps/slice.ts:11]:
```suggestion
  recentDappIds: string[]
```

File: src/dapps/slice.ts [hunk 3]
```diff
@@ -8,16 +8,15 @@ export interface State {
...
   dappListApiUrl: string | null
-  // TODO: update type of recentDapps and activeDapp to be string
   dappsList: Dapp[]
   dappsListLoading: boolean
   dappsListError: string | null
   dappsCategories: DappCategory[]
   dappConnectInfo: DappConnectInfo
   dappFavouritesEnabled: boolean
-  favouriteDapps: Dapp[]
+  favouriteDapps: string[]
```
Comment (jeanregisser) [inline src/dapps/slice.ts:19]:
```suggestion
  favouriteDappIds: string[]
```

File: src/dapps/selectors.ts (hunk 1 above)
Comment (kathaypacific) [reply src/dapps/selectors.ts]: lol should we actively try to avoid reduce? we haven't been consistent about this, i like reduce but i know you don't 😅

Comment (jeanregisser) [reply src/dapps/selectors.ts]: lol yeah I think our future selves will thank us for not abusing `reduce`

**Pull request merged**